repobee_sanitizer/_syntax.py

```diff
@@ -0,0 +1,127 @@
+"""Functions and constants relating to the syntax of repobee-sanitizer.
+
+.. module:: _syntaxhelpers
+    :synopsis: Functions and constants relating to the syntax of
+        repobee-sanitizer.
+"""
+import enum
+import pathlib
+from typing import List
+
+import re
+import repobee_plug as plug
+
+from repobee_sanitizer import _fileutils
+
+
+class Markers(enum.Enum):
+    """The markers that define sanitizer blocks."""
+
+    START = "REPOBEE-SANITIZER-START"
+    REPLACE = "REPOBEE-SANITIZER-REPLACE-WITH"
+    END = "REPOBEE-SANITIZER-END"
+
+
+def check_syntax(lines: List[str]) -> None:
+    """Checks if the input adheres to proper sanitizer syntax using proper
+    markers:
+
+    The grammar is defined below in EBNF-like manner. Note that the syntax is
+    not context- free as prefixes are defined on a per-block basis, so it's not
+    possible to perfectly describe the syntax with EBNF.
+
+    .. code-block:: raw
+            FILE ::=
+                MARKERLESS_LINE* ((BLOCK | PREFIXED_BLOCK) MARKERLESS_LINE*)+
+            BLOCK ::=
+                START_MARKER
+                MARKERLESS_LINE*
+                (REPLACE_MARKER
+                MARKERLESS_LINE*)?
+                END_MARKER
+            PREFIXED_BLOCK ::=
+                PREFIX START_MARKER
+                MARKERLESS_LINE*
+                (PREFIX REPLACE_MARKER
+                (PREFIX MARKERLESS_LINE)*)?
+                PREFIX END_MARKER
+            START_MARKER ::= "REPOBEE-SANITIZER-START\n"
+            REPLACE_MARKER ::= "REPOBEE-SANITIZER-REPLACE-WITH\n"
+            END_MARKER ::= "REPOBEE-SANITIZER-END\n"
+            MARKERLESS_LINE ::= line without sanitizer markers
+            PREFIX ::= a sequence of characters that is defined for each block
+                as any sequence that appears before the START_MARKER of that
+                particular PREFIXED_BLOCK.
```

Comment:
Two things here.

1. The contents of the block are too indented, 4 spaces should be removed from each line (i.e. dedent by 1 indentation). That's probably my fault, I didn't really check the indentation when I wrote the suggestion.
2. There must be a blank line after `.. code-block:: raw`, or `FILE ::=` will be interpreted as an option for the `code-block` directive.

Reply:
1. Thought about it but it looked wierd. However, by following 2 and adding a blank line, it now looks good

Reply:
Perfect.